app/services/RoutingService.scala

```diff
@@ -44,20 +46,33 @@ class RoutingService @Inject() (appConfig: AppConfig, messageConnector: MessageC
           officeOfDeparture(rootXml)
         }
 
-        officeEither.map {
+        officeEither.flatMap {
           office =>
-            if(office.value.startsWith("XI")) {
-              logger.info("routing to NI")
-              messageConnector.post(xml.toString(), appConfig.eisniUrl, appConfig.eisniBearerToken)
-            }
-            else {
-              logger.info("routing to GB")
-              messageConnector.post(xml.toString(), appConfig.eisgbUrl, appConfig.eisgbBearerToken)
+            getEisDetails(office, request.channel).map {
+              details => messageConnector.post(xml.toString(), details.url, details.token)
```

Comment:
I know this is existing functionality and not part of this PR, but passing the url and token into the connector feels like a smell (the connector is the thing that should know how to connect to its downstream; its callers shouldn't need to know implementation-level details like URLs).

Is it worth (maybe in a separate PR) refactoring so we either tell the connector to send to GB or NI and it knows URLs and tokens; or having 2 connectors and choosing which one to call?

Reply:
👍

Reply:
Could you create a tech dept ticket for this please Rob?

Reply:
I've already included this change @crystalmark